connectors/sources/generic_database.py

```diff
@@ -47,15 +47,33 @@ def is_wildcard(tables):
     return tables in (WILDCARD, [WILDCARD])
 
 
-def map_column_names(column_names, schema=None, table=None):
+def map_column_names(column_names, schema=None, tables=None):
     prefix = ""
     if schema and len(schema.strip()) > 0:
         prefix += schema.strip() + "_"
-    if table and len(table.strip()) > 0:
-        prefix += table.strip() + "_"
+    if tables and len(tables) > 0:
+        prefix += f"{'_'.join(sorted(tables))}_"
     return [f"{prefix}{column}".lower() for column in column_names]
 
 
+def hash_id(tables, row, primary_key_columns):
+    """Generates an id using table names as prefix in sorted order and primary key values.
+
+    Example:
+        tables: table1, table2
+        primary key values: 1, 42
+        table1_table2_1_42
+    """
+
+    if not isinstance(tables, list):
+        tables = [tables]
+
+    return (
+        f"{'_'.join(sorted(tables))}_"
+        f"{'_'.join([str(pk_value) for pk in primary_key_columns if (pk_value := row.get(pk)) is not None])}"
```

Comment:
The original implementation is not `join`. E.g. there are 2 columns `foo` and `bar` forming the primary key. Assuming the value for `foo` is `a`, and `bar` is `b`. Previously the id will be `database_schema_table_a_b_`, now it will be `database_schema_table_a_b`

Reply:
Yes, I think in last _ is not required in ID. So, I have change ID in both the document with sync rules and without sync rules. So it is uniform in both the cases.